[Enhancement] BE webserver support SSL #22971

LuGuangming opened the pull request:
## Proposed changes

Issue Number: close #22971 

<!--Describe your changes.-->

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

LuGuangming committed:
[Enhancement] BE webserver support SSL #22971

yiguolei commented on be/src/http/ev_http_server.h:
```diff
@@ -58,6 +63,8 @@ class EvHttpServer {
 private:
     Status _bind();
     HttpHandler* _find_handler(HttpRequest* req);
+    void _Init();
```
_Init()---> _init_ssl()
_ServerSetCerts(); --> _server_set_certs()

yiguolei commented on be/src/http/ev_http_server.cpp:
```diff
@@ -146,6 +202,12 @@ void EvHttpServer::stop() {
         }
         _event_bases.clear();
     }
+    if (m_ctx != NULL) {
```
!= nullptr

Hastyshell committed:
[feature](load) refactor CSV reading process during scanning, and support enclose and escape for stream load (#22539)

 ## Proposed changes

Refactor thoughts: close #22383
Descriptions about `enclose` and `escape`: #22385

## Further comments

2023-08-09: 
It's a pity that experiment shows that the original way for parsing plain CSV is faster. Therefor, the refactor is only applied on enclose related code. The plain CSV parser use the original logic.

Fallback of performance is unavoidable anyway. From the `CSV reader`'s perspective, the real weak point may be the write column behavior, proved by the flame graph.
 
Trimming escape will be enable after fix: #22411 is merged

Cases should be discussed: 

1. When an incomplete enclose appears in the beginning of a large scale data, the line delimiter will be unreachable till the EOF, will the buffer become extremely large?
2. What if an infinite line occurs in the case? Essentially,  `1.` is equivalent to this.  

Only support stream load as trial in this PR, avoid too many unrelated changes. Docs will be added when `enclose` and `escape` is available for all kinds of load.

irenesrl committed:
[Chore](docs)Add SSL Faq (#22956)

zhangstar333 committed:
[bug](if) fix if function not handle const nullable value (#22823)

fix if function not handle const nullable value

jacktengg committed:
[fix](datastream sender) fix wrong result of BUCKET_SHFFULE_HASH_PARTITIONED shuffle (#22973)

fix wrong result of BUCKET_SHFFULE_HASH_PARTITIONED shuffle

hubgeter committed:
[fix](createTableStmt)fix bug that createTableStmt toSql (#22750)

Issue Number: https://github.com/apache/doris/issues/22749

BiteTheDDDDt committed:
[Chore](excution) change some log fatal and dcheck to exception (#22890)

change some log fatal and dcheck to exception

xingyingone committed:
[Bug](aggregation)fix for map_agg when columns[1] is nullable (#22932)

In the map_agg handler function, added the judgment on columns[1]->is_nullable()

airborne12 committed:
[Fix](inverted index) fix non-index match function core (#22959)

kaijchen committed:
[refactor](load) change segcompaction worker interface (#22928)

morrySnow committed:
[fix](Nereids) type check could not work when root node is table or file sink (#22902)

type check could not work because no expression in plan.
sink and scan have no expression at all. so cannot check type.
this pr add expression on logical sink to let type check work well

BiteTheDDDDt committed:
[Bug](exchange) init _instance_to_rpc_ctx on register_sink (#22976)

init _instance_to_rpc_ctx on register_sink

Commit:
[FIX](map)insert into doris table with array/map type by local tvf (#22955)

jackwener committed:
[chore](Nereids): remove useless code (#22960)

LemonLiTree committed:
[fix](analysis) fix error msg #22950

Mryange committed:
[performance](executor) optimize time_round function only one arg (#22855)

zhangstar333 committed:
[refactor](parquet)change decimal type export as fixed-len-byte on parquet write (#22792)

before the parquet write export decimal as byte-binary,
but can't be import those fied to Hive.
Now, change to export decimal as fixed-len-byte-array in order to import hive directly.

freemandealer committed:
[fix](load) expose error root cause msg for load (#22968)

Currently, we only return ambiguous "INTERNAL ERROR" to the user when
load. This commit will no more hide the root cause.

Signed-off-by: freemandealer <freeman.zhang1992@gmail.com>

mrhhsg committed:
[feature](agg) Make 'map_agg' support array type as value (#22945)

Mryange committed:
[fix](function) fix str_to_date with specific format #22981

zddr committed:
[feature](auth)support Col auth (#22629)

support GRANT privilege[(col1,col2...)] [, privilege] ON db.tbl TO user_identity [ROLE 'role'];

dutyu committed:
[Enhancement](multi-catalog) add a BE selection strategy for hdfs short-circuit-read. (#22697)

Sometimes the BEs will be deployed on the same node with DataNode, so we can use a more reasonable BE selection policy to use the hdfs short-circuit-read as much as possible.

LuGuangming committed:
[Enhancement] BE webserver support SSL #22971
